src/server/plugins/internal/firestore.ts

```diff
@@ -63,13 +78,59 @@ const listenToTransactions = (server: Server, options: Options): (() => void) =>
     .onSnapshot((querySnapshot) => {
       querySnapshot.docChanges().forEach((change) => {
         if (change.type === 'added' && transactionHandlers.onCreate) {
-          transactionHandlers.onCreate(server, { id: change.doc.id, ...change.doc.data() })
-
+          transactionHandlers.onCreate(server, {
+            id: change.doc.id,
+            ...change.doc.data(),
+          })
         } else if (change.type === 'modified' && transactionHandlers.onUpdate) {
-          transactionHandlers.onUpdate(server, { id: change.doc.id, ...change.doc.data() })
-
+          transactionHandlers.onUpdate(server, {
+            id: change.doc.id,
+            ...change.doc.data(),
+          })
         } else if (change.type === 'removed' && transactionHandlers.onRemove) {
-          transactionHandlers.onRemove(server, { id: change.doc.id, ...change.doc.data() })
+          transactionHandlers.onRemove(server, {
+            id: change.doc.id,
+            ...change.doc.data(),
```

Comment:
copy all properties could be dangerous in schema-free databases - there is a possibility to store data we don't want/need to store

Reply:
hi pawel - can you raise an issue for this - i just made style changes here and for the sake of separation of concerns I don't want to modify code not related to this ticket/PR